### ISSUE TITLE
Should the loop over multiple files happen in the generic read or the file-type specific reads?

### ISSUE THREAD

**Issue by bhazelton** [RadioAstronomySoftwareGroup/pyuvdata#694]
Right now the generic read doesn't loop and each file type specific read has similar code to do the looping. Is that the pattern we want or should we move (with a deprecation period) to doing the loop in the generic read?

**Comment by adampbeardsley** [comment]
Personally I think we want to do the looping in the specific reads because users may call them directly (and our logic in generic read isn't airtight enough to require people use the generic). My proposal would be:
- Create a function that each specific read calls, and it will pass `file_type` based on which function its in.
- That function handles the checking for list and looping.
- For each read call, the function will call the generic read, specifying the `file_type` to ensure it reaches the right specific one.

As I wrote that, I realized it would also be reasonable to require the user use the generic read since we do have the `file_type` keyword which would be equivalent to using the specific reads now.

And final suggestion - we actually do both. The first option will clean up the code in the short term, and not change API. But we could throw a deprecation warning that we will soon move to the second solution. Then when we're ready to do so, it's a straightforward switch.
